core/src/sumeragi/mod.rs

```diff
@@ -243,66 +205,54 @@ impl SumeragiHandle {
         let (control_message_sender, control_message_receiver) = mpsc::sync_channel(100);
         let (message_sender, message_receiver) = mpsc::sync_channel(100);
 
-        let skip_block_count = wsv.block_hashes.len();
-        let mut blocks_iter = (skip_block_count + 1..=block_count).map(|block_height| {
-            kura.get_block_by_height(block_height as u64).expect(
-                "Sumeragi should be able to load the block that was reported as presented. \
-                 If not, the block storage was probably disconnected.",
-            )
-        });
+        let blocks_iter;
+        let current_topology;
```

Comment:
why did you write it this way?

Reply:
I didn't want to hold view for the whole scope of the function.

Reply:
Alternative would be: 

```
let (blocks_iter, current_topology) = {
    // ...
    (blocks_iter, current_topology)
}
```

Which doesn't look much more readable to me.